modules/images/webp-uploads/helper.php

```diff
@@ -65,6 +65,45 @@ function webp_uploads_get_upload_image_mime_transforms() {
  * @return array|WP_Error An array with the file and filesize if the image was created correctly otherwise a WP_Error
  */
 function webp_uploads_generate_additional_image_source( $attachment_id, array $size_data, $mime, $destination_file_name = null ) {
+
+	/**
+	 * Filter to allow the generation of additional image sources, in which a defined mime type
+	 * can be transformed and create additional mime types for the file.
+	 *
+	 * Returning an image data array or WP_Error here effectively short-circuits the default logic to generate the image source.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array|null|WP_Error  $image         Image data {'path'=>string, 'file'=>string, 'width'=>int, 'height'=>int, 'mime-type'=>string} or null or WP_Error.
+	 * @param int                  $attachment_id The ID of the attachment from where this image would be created.
+	 * @param string               $image_size    The size name that would be used to create this image, out of the registered subsizes.
+	 * @param array                $size_data     An array with the dimensions of the image: height, width and crop {'height'=>int, 'width'=>int, 'crop'}.
+	 * @param string               $mime          The target mime in which the image should be created.
+	 *
+	 * @return array|null|WP_Error An array with the file and filesize if the image was created correctly otherwise a WP_Error
+	 */
+	$image = apply_filters( 'webp_uploads_pre_generate_additional_image_source', null, $attachment_id, 'full', $size_data, $mime );
```

Comment:
@mehulkaklotar 
> so are we going to make bydefault `full` for all image size in filter? because it doesn't really make sense as I tested it and it turns out all image size data is being worked on in that.

Ahh, sorry I had totally missed this. Yes, this absolutely shouldn't show `full` for any image size. We will need to provide the name of the image size where the `$size_data` comes from. We don't have access to that currently from this function, so let's add a new function parameter `$size_name` right before `$size_data`.

We then need to pass this new parameter accordingly everywhere that `webp_uploads_generate_additional_image_source()` is called. Can you please make that update here?

Reply:
I have added changes for this in the latest commit.

Reply:
Thanks @mehulkaklotar. Please review my new feedback in https://github.com/WordPress/performance/pull/318#pullrequestreview-969948455 - part of your new implementation isn't needed, as the one function by definition is only used for the `full` image size, so we don't need to try to find the matching size, it's always `full` in that scenario.